modules/spark/src/main/java/io/fluo/recipes/spark/FluoSparkHelper.java

```diff
@@ -0,0 +1,297 @@
+   * @param opts Bulk import options
+   */
+  public void bulkImportRcvToAccumulo(JavaPairRDD<RowColumn, Bytes> data, String accumuloTable,
+      BulkImportOptions opts) {
+
+    data = partitionForAccumulo(data, accumuloTable, opts);
+
+    JavaPairRDD<Key, Value> kvData = data.mapToPair(tuple -> {
+      RowColumn rc = tuple._1();
+      byte[] row = rc.getRow().toArray();
+      byte[] cf = rc.getColumn().getFamily().toArray();
+      byte[] cq = rc.getColumn().getQualifier().toArray();
+      byte[] val = tuple._2().toArray();
+      return new Tuple2<>(new Key(new Text(row), new Text(cf), new Text(cq), 0), new Value(val));
+    });
+    bulkImportKvToAccumulo(kvData, accumuloTable, opts);
+  }
+
+  /**
+   * Bulk import Key/Value data into Accumulo
+   *
+   * @param data Key/value data to import
+   * @param accumuloTable Accumulo table used for import
+   * @param opts Bulk import options
+   */
+  public void bulkImportKvToAccumulo(JavaPairRDD<Key, Value> data, String accumuloTable,
+      BulkImportOptions opts) {
+
+    Path tempDir = getTempDir(opts);
+    Connector conn = chooseConnector(opts);
+
+    try {
+      if (hdfs.exists(tempDir)) {
+        throw new IllegalArgumentException("HDFS temp dir already exists: " + tempDir.toString());
+      }
+      hdfs.mkdirs(tempDir);
+      Path dataDir = new Path(tempDir.toString() + "/data");
+      Path failDir = new Path(tempDir.toString() + "/fail");
+      hdfs.mkdirs(failDir);
+
+      // save data to HDFS
+      Job job = Job.getInstance(hadoopConfig);
+      AccumuloFileOutputFormat.setOutputPath(job, dataDir);
+      // must use new API here as saveAsHadoopFile throws exception
+      data.saveAsNewAPIHadoopFile(dataDir.toString(), Key.class, Value.class,
+          AccumuloFileOutputFormat.class, job.getConfiguration());
+
+      // bulk import data to Accumulo
+      log.info("Wrote data for bulk import to HDFS temp directory: {}", dataDir);
+      conn.tableOperations().importDirectory(accumuloTable, dataDir.toString(), failDir.toString(),
+          false);
+
+      // throw exception if failures directory contains files
+      if (hdfs.listFiles(failDir, true).hasNext()) {
+        throw new IllegalStateException("Bulk import failed!  Found files that failed to import "
+            + "in failures directory: " + failDir);
+      }
+      log.info("Successfully bulk imported data in {} to '{}' Accumulo table", dataDir,
+          accumuloTable);
+
+      // delete data directory
+      hdfs.delete(tempDir, true);
+      log.info("Deleted HDFS temp directory created for bulk import: {}", tempDir);
+    } catch (IOException | TableNotFoundException | AccumuloException | AccumuloSecurityException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  /**
+   * Optional settings for Bulk Imports
+   */
+  public static class BulkImportOptions {
+
+    public static BulkImportOptions EMPTY = new BulkImportOptions();
```

Comment:
DEFTAULT may be a better name... EMPTY kinda works for the current options, but it may not work for future options that have a non-empty default